source/docs/casper/developers/dapps/monitor-and-consume-events.md

```diff
@@ -1,664 +1,495 @@
+<Tabs>
 
-**Deploy events**
+<TabItem value="js" label="JavaScript">
 
-These are associated with Deploys on a node. Currently, only `DeployAccepted` event is emitted . Refer to the [Deploys](../../concepts/design/casper-design.md#execution-semantics-deploys) section to discover more about Deploys and their life cycles.
+```javascript
+const { EventStream, EventName } = require("casper-js-sdk")
 
-**Finality Signature event**
+const es = new EventStream("http://NODE_ADDRESS:9999/events/" + CHANNEL)
+es.start()
+es.subscribe(EventName.EVENT_NAME, eventHandler)
 
-This event indicates that the final approvals from validators are signed and further alterations to the block will not be allowed. Refer to the [consensus reached](../../concepts/design/casper-design.md#consensus-reached) section and [block finality](../../concepts/glossary/B.md#block-finality) 
-section to learn more about finality signatures.
+function eventHandler(event) {
+    console.log(event)
+}
+```
 
-**Main events**
+</TabItem>
 
-All the events other than `DeployAccepted` and `FinalitySignature` fall under this type. Those are ApiVersion, BlockAdded, DeployProcessed, DeployExpired, Fault, Step, and Shutdown events.
+<TabItem value="python" label="Python">
 
-### How to monitor the event stream?
-You can start watching the event stream details using a simple Curl call as in the below format:
+```python
+from pycspr import NodeClient, NodeConnection, NodeEventChannel, NodeEventType
 
-```bash
-curl -sN http://<HOST:PORT>/events/<ENDPOINT>
+def eventHandler(event):
+    print(event)
+
+client = NodeClient(NodeConnection(host = "NODE_ADDRESS", port_rpc = 7777))
+client.get_events(eventHandler, NodeEventChannel.CHANNEL, NodeEventType.EVENT_NAME)
 ```
 
-- `HOST` - The IP address of a peer on the network
-- `PORT` - The port 9999 (The default port number for nodes on Mainnet or Testnet)
-- `ENDPOINT` - The specific endpoint of the node which the event will be emitted
+</TabItem>
 
-Refer to the [serialization standard](../../concepts/serialization-standard.md) page to get details on required custom serializations and the [types](../json-rpc/types_chain.md) page to find definitions of the terms used in the event stream output.
+</Tabs>
 
-## Monitoring DeployAccepted Event
+*Note: You can find active online peers' `NODE_ADDRESS`es at [cspr.live](https://cspr.live/tools/peers) for mainnet and [testnet.cspr.live](https://testnet.cspr.live/tools/peers) for testnet.*
 
-You can start watching the event stream for the `DeployAccepted` event or any other events being emitted on this endpoint using the following command. Replace the `HOST` field with the `peer IP address`.
+Where `CHANNEL` is `main` for an event type of `ApiVersion`, `BlockAdded`, `DeployExpired`, `DeployProcessed`, `Fault` or `Step`. `CHANNEL` is `deploys` for `DeployAccepted`, and `CHANNEL` is `sigs` for `FinalitySignature`.
```

Comment:
Can you rephrase/restructure this? I think I know what is meant, but if you read that back to yourself I think you'll agree that it's as clear as mud on first reading. 
```suggestion
Where `CHANNEL` is set to;
 `main` for the following event types;
 `ApiVersion`, `BlockAdded`, `DeployExpired`, `DeployProcessed`, `Fault` or `Step`. 
`deploys` for event types;
 `DeployAccepted`, or 
`sigs` for event type `FinalitySignature`.
```